### PR TITLE
Add support for PowerShell Remoting sessions

### PR DESCRIPTION
**Pull request by Anthirian**
I have added support for PowerShell Remoting (also called WinRM) sessions to the PowerShellSessionLexer. These follow the format below.

```
[Hostname]: PS C:\> Get-ChildItem
```

I have added this by adding an optional non-capturing group to the regex in front of the traditional regex. I hope this follows all the rules and that I did not miss an edge case.

### CODE REVIEW

**Comment by Anteru** [comment]
Could you please add a test case for this? You can generate one using the `unittest` formatter, but I'd feel better about it with at least some basic test added.

**Comment by Anthirian** [comment]
Sure, I will add it soon

**Comment by Anthirian** [comment]
I have added the test case you requested. Please check to see if it makes sense to test it this way (I'm fairly new to unit testing).

File: pygments/lexers/shell.py [hunk 1]
```diff
@@ -772,7 +772,7 @@ class PowerShellSessionLexer(ShellSessionBaseLexer):
     mimetypes = []
 
     _innerLexerCls = PowerShellLexer
-    _ps1rgx = re.compile(r'^(PS [^>]+> )(.*\n?)')
+    _ps1rgx = re.compile(r'^((?:\[[^]]+\]: )?PS [^>]+> ?)(.*\n?)')
```
[the remark this answers is not in the document]
Comment (Anthirian) [reply pygments/lexers/shell.py:775]: Sorry for the delay, I didn't have time to spare on this until now.
This second question mark was actually to make the block in square brackets optional, allowing the regex to match both local and remote PowerShell sessions, without having to create a separate subclass for the remote case.

File: tests/test_shell.py [hunk 2]
```diff
@@ -163,6 +168,18 @@ def test_msdos_gt_only(lexer_msdos):
...
+
+def test_powershell_remoting_session(lexer_powershell_session):
+    fragment = u'[Long-NetBIOS-Hostname]: PS C:\\> Get-ChildItem\n'
+    tokens = [
+        (Token.Name.Builtin, u''),
+        (Token.Generic.Prompt, u'[Long-NetBIOS-Hostname]: PS C:\\> '),
+        (Token.Name.Builtin, u'Get-ChildItem'),
+        (Token.Text, u'\n')
+    ]
+    assert list(lexer_powershell_session.get_tokens(fragment)) == tokens
```
[the remark this answers is not in the document]
Comment (Anthirian) [reply tests/test_shell.py:180]: Okay I understand that's needed from a consistency point of view, however when I implemented this test for the remote sessions use case there was not one for local sessions at all. I could add it if you really wish, but I'd say it was forgotten when adding the current local `PowerShellSessionLexer` class.

[the remark this answers is not in the document]
Comment (Anteru) [reply tests/test_shell.py:180]: I'd prefer to have a test for both; and yes, test coverage wasn't great before :( I simply want to make sure new functionality doesn't break existing one.

File: pygments/lexers/shell.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Anteru) [reply pygments/lexers/shell.py:775]: Sorry, I wasn't clear :( I was referring to the one *after* the `>`, i.e. this one: 
```
^((?:\[[^]]+\]: )?PS [^>]+> ?)(.*\n?)
                            ^
```
Previously, that wasn't there. I understand why you want to make the first block optional, but that one I didn't quite get (and I just realized it's the _third_ `?`, not the second ...)

[the remark this answers is not in the document]
Comment (Anthirian) [reply pygments/lexers/shell.py:775]: Oh right, if I remember correctly it was due to the fact that sometimes the space after the `>` isn't printed, so I figured I'd make it optional. It might have just been an error in my tests though. Not sure about policies, but I'd say it doesn't hurt to keep it in there right?

File: tests/test_shell.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (Anthirian) [reply tests/test_shell.py:180]: Okay I'll add the local PowerShell sessions test as well.

File: pygments/lexers/shell.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Anteru) [reply pygments/lexers/shell.py:775]: I guess so :)

**Comment by Anteru** [comment]
Merged, thanks!
